consumer/shard_api.go

```diff
@@ -126,6 +127,14 @@ func ShardApply(ctx context.Context, claims pb.Claims, srv *Service, req *pc.App
 			}
 			key = allocator.ItemKey(s.KS, change.Upsert.Id.String())
 			ops = append(ops, clientv3.OpPut(key, change.Upsert.MarshalString()))
+
+			if change.PrimaryHints != nil {
```

Comment:
optional suggestion: it might be worth logging something here

IDK if there's any other validation we should be doing on the hints, but if we wanted to check the actual journals and offsets, then we'd need to do that here. IDK how much value there is in that, though, since if they're wrong we'll find out pretty quick once the shard starts.

Reply:
Yea, beyond running Validate (which I've added), I think that's out of scope for this API.